addons/product/models/product.py

```diff
@@ -572,7 +572,9 @@ def open_product_template(self):
                 'target': 'new'}
 
     def _prepare_sellers(self, params):
-        return self.seller_ids.filtered(lambda s: s.name.active).sorted(lambda s: (s.sequence, -s.min_qty, s.price))
+        return self.env['product.supplierinfo'].search(['|', ('product_id', '=', self.id),
+                                                        ('product_tmpl_id', '=', self.product_tmpl_id.id),
```

Comment:
Note that this should work since `len(self) == 1` thanks to:

https://github.com/odoo/odoo/blob/2c71cd239bbbbc41eaf539d5fd210d5bae008593/addons/product/models/product.py#L578